src/Microsoft.Health.SqlServer/DefaultSqlConnectionStringProvider.cs

```diff
@@ -21,11 +22,24 @@ public class DefaultSqlConnectionStringProvider : ISqlConnectionStringProvider
     public DefaultSqlConnectionStringProvider(IOptions<SqlServerDataStoreConfiguration> sqlServerDataStoreConfiguration)
     {
         _sqlServerDataStoreConfiguration = EnsureArg.IsNotNull(sqlServerDataStoreConfiguration?.Value, nameof(sqlServerDataStoreConfiguration));
+
+        if (_sqlServerDataStoreConfiguration.MaxPoolSize.HasValue)
+        {
+            EnsureArg.IsGt(_sqlServerDataStoreConfiguration.MaxPoolSize.Value, 0, nameof(_sqlServerDataStoreConfiguration.MaxPoolSize));
+            EnsureArg.IsLt(_sqlServerDataStoreConfiguration.MaxPoolSize.Value, SqlServerDataStoreConfiguration.MaxPoolSizeLimit, nameof(_sqlServerDataStoreConfiguration.MaxPoolSize));
+        }
     }
 
     /// <inheritdoc />
     public Task<string> GetSqlConnectionString(CancellationToken cancellationToken)
     {
-        return Task.FromResult(_sqlServerDataStoreConfiguration.ConnectionString);
+        string connectionString = _sqlServerDataStoreConfiguration.ConnectionString;
+
+        if (_sqlServerDataStoreConfiguration.MaxPoolSize.HasValue && !connectionString.Contains(SqlServerDataStoreConfiguration.MaxPoolSizeName, StringComparison.OrdinalIgnoreCase))
+        {
+            return Task.FromResult($"{connectionString};{SqlServerDataStoreConfiguration.MaxPoolSizeName}={_sqlServerDataStoreConfiguration.MaxPoolSize.Value};");
```

Comment:
can we yse SqlConnectionStringBuilder class to do this instead of contactination?

Reply:
Done